Dashboard/app/js/lib/components/devices/AssignmentsEditView/screens/AssignDatapoints/index.jsx

```diff
@@ -20,7 +20,9 @@ export default class AssignDatapoints extends React.Component {
     const { devices, datapointAssignments } = this.context.data;
 
     const deviceData = devices.find(device => device.id === selectedDeviceId);
-    const datapointAssignment = datapointAssignments.find(dp => dp.deviceId === selectedDeviceId);
+    const datapointAssignment = datapointAssignments.find(
+      dp => dp.deviceId === parseInt(selectedDeviceId, 10)
```

Comment:
Instead of doing this I think its much clearer to set the property to a number. i.e. 

```
AssignDatapoints.propTypes = {
  selectedDeviceId: PropTypes.number.isRequired,
};
```

Reply:
Yeah, But in #3404 it's already converted to a number, this is like a minor hack to make sure this PR works